hcn/hcnloadbalancer.go

```diff
@@ -333,6 +351,15 @@ func AddLoadBalancer(endpoints []HostComputeEndpoint, flags LoadBalancerFlags, p
 		Flags: flags,
 	}
 
+	if additionalParams != nil {
+		switch additionalParams[0] {
```

Comment:
I think the additionalParams makes more sense as an enum. Were you trying to avoid breaking anyone currently calling this function?

Reply:
Yes, that was my reasoning for the variadic trailing parameter. The other option like @jterry75 mentioned is to add an overload to this function, but we'd have to name it something else because go doesn't allow overloading. Not sure which route we should take

Reply:
Oh also I made additionalParams a  variadic string parameter so that it could be used for other input in the future (you can pass as many strings in as you want to additionalParams). Right now we'd just be looking at the first one, but we could accept more string input through this parameter if needed.

Reply:
I'm okay with breaking people to be honest. I suspect you and @jterry75 might feel differently about that though. Otherwise, I think the overloading and calling with defaults seems best.

Reply:
@jterry75  I'll let you make the call. Add a new enum parameter or break existing callers?

Reply:
Its your API but at a bare minimum I would like to see the cases documented. And we should loop `additionalParams` and handle all of them. Rater than the assumption that it is of size 1.